### PR TITLE
feat: add RSD instrumentation service support

### PR DESCRIPTION
**Pull request by michal-przytarski**

### CODE REVIEW

**Review by dmissmann — CHANGES_REQUESTED** [review]
Great, they moved those things as they are to `dtservicehub` 🙂

File: ios/listdevices.go [hunk 1]
```diff
@@ -82,6 +82,10 @@ func NewReadDevices() ReadDevicesType {
 	return data
 }
 
+func (device *DeviceEntry) SupportsRsd() bool {
```
[the remark this answers is not in the document]
Comment (michal-przytarski) [reply ios/listdevices.go:87]: done :D [22ba1b1](https://github.com/danielpaulus/go-ios/pull/396/commits/22ba1b15953b5be15652be61723f8ccd4d23a6ed)

File: ios/instruments/helper.go [hunk 2]
```diff
@@ -24,6 +25,10 @@ func (p loggingDispatcher) Dispatch(m dtx.Message) {
 }
 
 func connectInstruments(device ios.DeviceEntry) (*dtx.Connection, error) {
+	if device.SupportsRsd() {
```
[the remark this answers is not in the document]
Comment (michal-przytarski) [reply ios/instruments/helper.go:28]: done :D [5dfc702](https://github.com/danielpaulus/go-ios/pull/396/commits/5dfc7023f78fc0598c272b0e2c6739a7dfa8bdf9)

**Review by dmissmann — APPROVED** [review]
thanks! 🙇‍♂️

**Comment by michal-przytarski** [comment]
Hi
What should we do with this PR? Pretty much all the changes are already merged. Only refactor left.
Do you want to merge it, or should I close it?

**Comment by dmissmann** [comment]
> Do you want to merge it, or should I close it?

I would still merge it, but I can't. It says `This branch is out-of-date with the base branch` on my end and I assume that just means that you have to merge `main` into your branch, or rebase onto `main`

**Pull request merged**